### PR TITLE
Table editor filter and search UI fixes

### LINKED ISSUE #475: Qt TableEditor ignores factory setting for filter
Setting `filter=` in the factory is ignored by the editor, so not filter is selected initially.

### PR DESCRIPTION
**Pull request by corranwebster**
Fixes enabled state of Wx TableEditor search button (looks like error from bad merge).  Also fixes #475 and some other minor errors in the TableEditors.

### CODE REVIEW

File: traitsui/wx/table_editor.py [hunk 1]
```diff
@@ -1030,13 +1030,13 @@ def _update_toolbar(self, has_selection):
                         deletable(item) for item in self.selected_items
                     )
 
-                toolbar.search.enabled = toolbar.add.enabled = True
+                toolbar.search.enabled = no_filter
```
Comment (corranwebster) [inline traitsui/wx/table_editor.py]: This is not right.

File: traitsui/tests/editors/test_table_editor.py [hunk 2]
```diff
@@ -254,6 +271,27 @@ def test_table_editor():
...
+        values=[ListItem(value=str(i**2)) for i in range(10)]
+    )
+
+    with store_exceptions_on_all_threads():
+        ui = object_list.edit_traits(view=filtered_view)
+        gui.process_events()
+
+        print('here')
+        filter = ui.get_editors('values')[0].filter
+        print(filter)
```
Comment (jwiggins) [inline traitsui/tests/editors/test_table_editor.py]: Debugging leftovers?

File: traitsui/table_filter.py [hunk 3]
```diff
@@ -412,7 +412,10 @@ class GenericTableFilterRuleEnabledColumn(ObjectColumn):
     def get_value(self, object):
         """ Returns the traits editor of the column for a specified object.
         """
-        return ['', '==>'][object.enabled]
+        if hasattr(object, 'enabled') and object.enabled:
+            return u'\u2714'
```
Comment (jwiggins) [inline traitsui/table_filter.py]: What's that?

File: traitsui/wx/table_editor.py (hunk 1 above)
Comment (jwiggins) [reply traitsui/wx/table_editor.py]: Is it fixed now, or is it still wrong?

File: traitsui/table_filter.py (hunk 3 above)
Comment (mdickinson) [reply traitsui/table_filter.py]: Maybe it would be better spelled as `u'\N{HEAVY CHECK MARK}'`?

File: traitsui/wx/table_editor.py (hunk 1 above)
Comment (corranwebster) [reply traitsui/wx/table_editor.py]: Still wrong - haven't gotten back to it since looking over yesterday.

File: traitsui/table_filter.py (hunk 3 above)
Comment (corranwebster) [reply traitsui/table_filter.py]: Yep - will change.

**Pull request merged**